### ISSUE TITLE
[wordify] Wordify BP Freemarker template error when initializing site in Delivery

### ISSUE THREAD

**Issue by Euquimides** [craftercms/craftercms#6506]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

Initializing a site in Delivery based on the Wordify BP, will give a Freemarker template error. 

### Steps to reproduce

Steps:

1. Crete a new project or site based on Wordify BP. 
2. Publish the site or project.
3. Initialize site or project in Delivery. 
4. See the issue

### Relevant log output

```shell
[INFO] 2024-02-01T17:25:02,064 [pool-15-thread-1] [wordify] [context.SiteContext] | </Initializing context site: wordify> 
[INFO] 2024-02-01T17:25:02,064 [pool-15-thread-1] [wordify] [context.SiteContext] | -------------------------------------------------- 
01-Feb-2024 17:28:44.053 SEVERE [http-nio-9080-exec-3] freemarker.log._JULLoggerFactory$JULLogger.error Error executing FreeMarker template
	FreeMarker template error:
	Java method "org.craftercms.engine.service.impl.SiteItemServiceImpl.getSiteItem(String)" threw an exception when invoked on org.craftercms.engine.service.impl.SiteItemServiceImpl object "org.craftercms.engine.service.impl.SiteItemServiceImpl@442d8c52"; see cause exception in the Java stack trace.

	----
	FTL stack trace ("~" means nesting-related):
	- Failed at: #assign levelDescriptor = siteItemSer...  [in template "templates/web/fragments/header.ftl" at line 3, column 1]
	- Reached through: #include "/templates/web/fragments/he...  [in template "templates/web/entry.ftl" at line 12, column 7]
	----

	Java stack trace (for programmers):
	----
	freemarker.core._TemplateModelException: [... Exception message was already printed; see it above ...]
```

### Screenshots and/or videos

https://github.com/craftercms/craftercms/assets/90878288/de85123d-282f-4575-b0ec-4e6ec1c70a39

**Comment by rart** [comment]
@jvega190 we want to remove the usage of the levelDescriptor retrieval using the problematic expression. The idea would be to preferably use the model or retrieve the level descriptor in a different way.

**Comment by Euquimides** [comment]
Hi! Updated and rechecked this issue. Fix verified for Crafter v4.1.4, v4.2.0. Thank you! Closing this ticket.
